htdocs/editor_tab.js

```diff
@@ -195,8 +195,11 @@ var editor = function () {
             return Promise.cast(tree_controller_.find_next_copy_name(username_, new_notebook_prefix_ + '1'))
                 .then(shell.new_notebook.bind(shell))
                 .then(function(notebook) {
-                    tree_controller_.set_visibility(notebook.id, true);
-                    that.star_notebook(true, {notebook: notebook, make_current: true, version: null});
+                    return tree_controller_.set_visibility(notebook.id, true).then(function(x) {
+                        return notebook;
+                      });
+                }).then(function(notebook) {
+                  return that.star_notebook(true, {notebook: notebook, make_current: true, version: null});
                 });
```

Comment:
Is this part of the fix? These were intentionally in parallel before, I will probably change to `Promise.all()` before merging.